### ISSUE TITLE
Cifar and Imagenet examples use virtual batch size for batch normalization 

### ISSUE THREAD

**Issue by PForet** [google/flax#319]
So far, the Cifar and Imagenet examples don't sync the batch statistics across replicas during batch normalization. For multi-GPU training, this is equivalent to using smaller virtual batches than the total batch size advertised. 
This leads to less latency and better generalization (and is not uncommon in other implementations), so this behavior should be kept, but it should be more explicit. Maybe by using a flag for synchronization that is turned off by default? Or at least by pointing out the `axis_name` argument of the batchnorm constructor in some comment.

**Comment by AlexeyG** [comment]
Hi Pierre,

Thank you for opening the issue. I think you make a good point - we should be explicit about this decision in our implementations. One thing we'd like to do is to keep the examples relatively simple, so I'm not so much in favour adding a flag that changes the training time behaviour.

But I think your idea with adding a comment somewhere around the BN constructor is great. Would you like to submit a PR for this?

**Comment by marcvanzee** [comment]
It appears that we are now actually syncing the batch statistics across replicas during batch normalization, see here: https://github.com/google/flax/blob/811084f48fd6f1edb2cd4775e375df8edc8a06eb/examples/imagenet/train.py#L214

Closing this.
